### ISSUE TITLE
Add support for certificate authentication

### ISSUE THREAD

**Issue by garrytrinder** [ValoIntranet/doctor#34]
In [CLI for Microsoft 356 v3.4](https://dev.to/microsoft365/cli-for-microsoft-365-v3-4-2d14) we have simplified the `m365 login` command for certificate authentication and provided options for passing `--clientId` and `--tenant` into the login command as an alternative to our environment variables approach to use your own Azure AD identity for authentication.

It would be useful if we could extend the `publish` command as below to incorporate certificate authentication.

```
doctor publish --auth certificate --certificateBase64Encoded MIII2QIBAzCCCJ8GCSqGSIb3DQEHAaCCCJAEg...eX1N5AgIIAA== --appId 31359c7f-bd7e-475c-86db-fdb8c937548c --tenant 31359c7f-bd7e-475c-86db-fdb8c937548a --url https://contoso.sharepoint.com/sites/docs
```

This would extend `--auth` with a new option called `certificate` and introduce the basic options, `--appId`, `--tenant` and  `--certificateBase64Encoded`.

> `tenant` can be omitted if using a multi-tenant app registration

**Comment by estruyf** [comment]
Thanks @garrytrinder, that would be a good enhancement!

**Comment by estruyf** [comment]
This enhancement is now available in the `v1.6.0` beta version. Feel free to test it out.

Documentation how to use it has been added in the readme file from the DEV branch: [https://github.com/ValoIntranet/doctor/tree/dev#certificate-authentication](https://github.com/ValoIntranet/doctor/tree/dev#certificate-authentication)

**Comment by estruyf** [comment]
Version `1.6.0` is released with this issue included.
